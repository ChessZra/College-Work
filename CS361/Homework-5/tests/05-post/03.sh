#!/bin/bash

PORT=$@

REQUEST1=$'POST /write HTTP/1.1\r
Content-Length: 98\r
\r
Write a lot of data on the first time to make sure the second write is not combined with the first'

REQUEST2=$'POST /write HTTP/1.1\r
Content-Length: 17\r
\r
Write little data'

REQUEST3=$'GET /read HTTP/1.1\r\n\r\n'


printf "$REQUEST1" | nc -N 127.0.0.1 $PORT
printf "\n"
printf "$REQUEST2" | nc -N 127.0.0.1 $PORT
printf "\n"
printf "$REQUEST3" | nc -N 127.0.0.1 $PORT
printf "\n"
