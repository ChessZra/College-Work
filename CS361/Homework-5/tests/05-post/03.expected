HTTP/1.1 200 OK
Content-Length: 98

Write a lot of data on the first time to make sure the second write is not combined with the first
HTTP/1.1 200 OK
Content-Length: 17

Write little data
HTTP/1.1 200 OK
Content-Length: 17

Write little data
